### ISSUE TITLE
[request] Support Custom Route attributes

### ISSUE THREAD

**Issue by ProbablePrime** [michalkvasnicak/routex#19]
Currently when the routes are instantiated into Route objects only a subset of the original route object copied across into a Route instance.

I'm not using react so the component attribute is of no use.
I'd like to be able to keep any custom attributes on the route e.g
```
const routes = [
	{
		path: '/',
		specialSauce: 1
	}
];
```
These are instead silently discarded.

For a router that claims to not require React the definition of the Route class kind of pushes react.

**Comment by michalkvasnicak** [comment]
How it should work if you have nested routes? Should it extend parent routes with custom attributes from nested routes?

What about little change to your example and make it something like:

```js
const routes = [
    {
        path: '/',
        attrs: {
            specialSauce: 1
        }
    }
];
```

What do you think ? :)

**Comment by ProbablePrime** [comment]
I hadn't thought about nested routes, extension sounds the best way to deal with it yes.

As for moving custom attributes inside their own object I agree. It makes it easier to make backwards compatible changes to routex and to people's existing route definitions. 

thanks!

**Comment by michalkvasnicak** [comment]
Released as `1.0.0-alpha.20`

**Comment by ProbablePrime** [comment]
You are the best. Thanks so much! That was so fast.
